### PR TITLE
Fix clippy & Cargo update

### PR DESCRIPTION
**Pull request by Boog900**

### CODE REVIEW

File: deny.toml [hunk 1]
```diff
@@ -120,6 +120,7 @@ allow = [
 
     # Font licenses.
     "Unicode-DFS-2016",     # https://spdx.org/licenses/Unicode-DFS-2016.html
+    "Unicode-3.0",          # https://spdx.org/licenses/Unicode-3.0.html
```
Comment (hinto-janai) [inline deny.toml:123]: Do you know which crate is pulling this in?

File: rpc/json-rpc/src/id.rs [hunk 2]
```diff
@@ -182,20 +182,20 @@ macro_rules! impl_u {
         $(
             impl From<$u> for Id {
                 fn from(u: $u) -> Self {
-                    Self::Num(u as u64)
+                    Self::Num(u64::from(u))
```
Comment (hinto-janai) [inline rpc/json-rpc/src/id.rs:185]: Oops, original impl took `i8-i64` and `f32/f64` as input which is why `as` was used.

**Review by hinto-janai — COMMENTED** [review]
Sorry, the `#![cfg(test)]` at [`rpc/json-rpc/src/tests.rs:3`](https://github.com/Cuprate/cuprate/actions/runs/9503481103/job/26193991437#step:11:62) should be removed.

File: deny.toml (hunk 1 above)
Comment (Boog900) [reply deny.toml:123]: heed pulls in `url` which pulls in crates from here: https://github.com/unicode-org/icu4x

**Review by hinto-janai — APPROVED** [review]

**Pull request merged**